windows-server-container-tools/New-ContainerHost/New-ContainerHost.ps1

```diff
@@ -1209,6 +1216,121 @@ Find-DockerImages
+    }
+
+    $dockerDaemonScript = "$dockerData\runDockerDaemon.cmd"
+
+    New-DockerDaemonRunText | Out-File -FilePath $dockerDaemonScript -Encoding ASCII
+
+    if (Test-Nano)
+    {
+        Write-Output "Creating scheduled task action..."
+        $action = New-ScheduledTaskAction -Execute "cmd.exe" -Argument "/c $dockerDaemonScript > $dockerLog 2>&1"
+
+        Write-Output "Creating scheduled task trigger..."
+        $trigger = New-ScheduledTaskTrigger -AtStartup
+
+        Write-Output "Creating scheduled task settings..."
+        $settings = New-ScheduledTaskSettingsSet -Priority 5
+
+        Write-Output "Registering Docker daemon to launch at startup..."
+        Register-ScheduledTask -TaskName $global:DockerServiceName -Action $action -Trigger $trigger -Settings $settings -User SYSTEM -RunLevel Highest | Out-Null
+
+        Write-Output "Launching daemon..."
+        Start-ScheduledTask -TaskName $global:DockerServiceName
+    }
+    else
+    {
+        if (Test-Path "$($env:SystemRoot)\System32\nssm.exe")
+        {
+            Write-Output "NSSM is already installed"
+        }
+        else
+        {
+            Get-Nsmm -Destination "$($env:SystemRoot)\System32" -WorkingDir "$env:temp"
+        }
+
+        Write-Output "Configuring NSSM for $global:DockerServiceName service..."
+        Start-Process -Wait "nssm" -ArgumentList "install $global:DockerServiceName $($env:SystemRoot)\System32\cmd.exe /s /c $dockerDaemonScript < nul"
+        Start-Process -Wait "nssm" -ArgumentList "set $global:DockerServiceName DisplayName Docker Daemon"
+        Start-Process -Wait "nssm" -ArgumentList "set $global:DockerServiceName Description The Docker Daemon provides management capabilities of containers for docker clients"
+        # Pipe output to daemon.log
+        Start-Process -Wait "nssm" -ArgumentList "set $global:DockerServiceName AppStderr $dockerLog"
+        Start-Process -Wait "nssm" -ArgumentList "set $global:DockerServiceName AppStdout $dockerLog"
+        # Allow 30 seconds for graceful shutdown before process is terminated
+        Start-Process -Wait "nssm" -ArgumentList "set $global:DockerServiceName AppStopMethodConsole 30000"
+
+        Start-Service -Name $global:DockerServiceName
+    }
+
+    #
+    # Waiting for docker to come to steady state
+    #
+    Wait-Docker
+
+    Write-Output "The following images are present on this machine:"
+    foreach ($image in (Get-DockerImages))
+    {
+        Write-Output "    $image"
+    }
+    Write-Output ""
+}
+
+
+function
+New-DockerDaemonRunText
+{
+    return @"
+
+@echo off
+set certs=%ProgramData%\docker\certs.d
+
+if exist %ProgramData%\docker (goto :run)
+mkdir %ProgramData%\docker
+
+:run
+if exist %certs%\server-cert.pem (if exist %ProgramData%\docker\tag.txt (goto :secure))
+
+dockerd daemon -H nipe:// 
+goto :eof
+
+:secure
+dockerd daemon -H 0.0.0.0:2376 --tlsverify --tlscacert=%certs%\ca.pem --tlscert=%certs%\server-cert.pem --tlskey=%certs%\server-key.pem
```

Comment:
I suggest to add `-H npipe:// -H tcp://0.0.0.0:2376` here.
Probably my PR #169 will be hard to rebase after these changes, I can close it.